### PR TITLE
new plugin ShowAllRoles: show all roles in simplified profiles

### PR DESCRIPTION
**Pull request by Luna-devv**
Show all members' roles on the pop-out again.

| Before - Limited to two rows  | After - Show all roles |
| ------------- | ------------- |
| image  | image  |

### CODE REVIEW

File: src/plugins/showAllRoles/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export default definePlugin({
+    name: "ShowAllRoles",
+    description: "Show all members' roles on the pop-out again.",
+    authors: [Devs.Luna],
+    patches: [
+        {
+            find: ".Messages.VIEW_ALL_ROLES",
+            replacement: {
+                match: /return\s+null\s*!=\s*(\w+)\s*\?\s*(\w+)\.slice\(0,\s*\1\)\s*:\s*\2/,
```
Comment (Masterjoona) [inline src/plugins/showAllRoles/index.ts:18]: you can simplify this to be `return null!=.{13,17}:(\i)`

Comment (Luna-devv) [reply src/plugins/showAllRoles/index.ts]: thank you <3

**Comment by Vendicated** [comment]
add a README please

File: src/plugins/showAllRoles/index.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (c) 2024 Vendicated and contributors
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ */
+
+import { Devs } from "@utils/constants";
+import definePlugin from "@utils/types";
+
+export default definePlugin({
+    name: "ShowAllRoles",
+    description: "Show all members' roles on the pop-out again.",
```
Comment (Vendicated) [inline src/plugins/showAllRoles/index.ts:12]:
```suggestion
    description: "Show all roles in new profiles",
```

File: src/plugins/showAllRoles/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    authors: [Devs.Luna],
+    patches: [
+        {
+            find: ".Messages.VIEW_ALL_ROLES",
+            replacement: {
+                match: /return null!=.{13,17}:(\i)/,
+                replace: "return $1"
```
Comment (Vendicated) [inline src/plugins/showAllRoles/index.ts:19]:
```suggestion
                match: /return null!=\i(?=\?\i\.slice)/,
                replace: "return false"
```

**Comment by Vendicated** [comment]
looks good!

**Pull request merged**